EventFilter/ESRawToDigi/src/ESUnpacker.cc

```diff
@@ -242,10 +257,12 @@ void ESUnpacker::interpretRawData(int fedId, const FEDRawData & rawData, ESRawDa
     } else if (head == 6) {
       optoBC = (*word >> 32) & m16;
       optoEC = (*word >> 48) & m8;      
+
+      opto = enableOptoRX[iopto];
       if (opto==0) ESDCCHeader.setOptoBC0(optoBC);
       else if (opto==1) ESDCCHeader.setOptoBC1(optoBC);
       else if (opto==2) ESDCCHeader.setOptoBC2(optoBC);
-      opto++;
+      iopto++;
```

Comment:
I can't tell if iopto can reach 3 or more. Please make sure it doesn't exceed 2, to protect against an out-of-bounds error:
```
if (iopto < 2)
   ++iopto;
```